components/nimbus/src/enrollment.rs

```diff
@@ -534,19 +536,19 @@ pub(crate) trait ExperimentMetadata {
 
 pub(crate) struct EnrollmentsEvolver<'a> {
     available_randomization_units: &'a AvailableRandomizationUnits,
-    targeting_helper: &'a NimbusTargetingHelper,
+    targeting_helper: Rc<Mutex<NimbusTargetingHelper>>,
```

Comment:
This doesn't need to be behind Rc<Mutex> since you're not handing out the Rc anywhere. You can keep a mut ref inside the EnrollmentsEvolver (which means you can mutate targeting_helper even if the EnrollmentsEvolver *isnt* mut).

```suggestion
    targeting_helper: &'a mut NimbusTargetingHelper,
```